htdocs/core/class/commonobject.class.php

```diff
@@ -4250,7 +4250,7 @@ function updateExtraField($key)
 
 		if (! empty($conf->global->MAIN_EXTRAFIELDS_DISABLED)) return 0;	// For avoid conflicts if trigger used
 
-        if (! empty($this->array_options) && !empty($this->array_options["options_$key"]))
+        if (! empty($this->array_options) )
```

Comment:
I think correct fix, should be 
if (! empty($this->array_options) && isset($this->array_options["options_$key"]))

Does this other suggestion work for your case ?